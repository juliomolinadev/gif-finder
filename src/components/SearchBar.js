import React, { useState } from "react";
import PropTypes from "prop-types";

const SearchBar = ({ setWord }) => {
	const [value, setValue] = useState("");

	const handleInputChange = (e) => {
		setValue(e.target.value);
	};

	const handleSetWord = (e) => {
		e.preventDefault();
		setWord(value);
		setValue("");
	};

	return (
		<form className="d-flex" onSubmit={handleSetWord}>
			<input
				id="searchInput"
				type="text"
				value={value}
				onChange={handleInputChange}
				className="form-control"
			/>

			<button className="btn btn-primary ms-2">Search</button>
		</form>
	);
};

SearchBar.propTypes = {
	setWord: PropTypes.func.isRequired
};

export default SearchBar;
