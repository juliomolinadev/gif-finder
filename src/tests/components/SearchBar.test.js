import React from "react";
import SearchBar from "../../components/SearchBar";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

describe("Tests in <SearchBar />", () => {
	const setWord = jest.fn();
	let wrapper;

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<SearchBar setWord={setWord} />);
	});

	test("Should be displayed correctly", () => {
		expect(wrapper).toMatchSnapshot();
	});

	test("Text input should change", () => {
		const input = wrapper.find("input");
		const value = "hola";

		input.simulate("change", { target: { value } });
		expect(wrapper.find("input").prop("value")).toBe(value);
	});

	test("Should not post the form", () => {
		wrapper.find("form").simulate("submit", { preventDefault() {} });
		expect(setWord).not.toHaveBeenCalled();
	});

	test("Should to call setWord and clear the input", () => {
		const value = "hola";

		wrapper.find("input").simulate("change", { target: { value } });
		wrapper.find("form").simulate("submit", { preventDefault() {} });

		expect(setWord).toHaveBeenCalled();
		expect(setWord).toHaveBeenCalledTimes(1);
		expect(setWord).toHaveBeenCalledWith(expect.any(String));
		expect(wrapper.find("input").prop("value")).toBe("");
	});
});
