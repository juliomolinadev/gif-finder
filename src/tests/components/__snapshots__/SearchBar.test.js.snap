// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <SearchBar /> Component should be displayed correctly 1`] = `
<form
  className="d-flex"
  onSubmit={[Function]}
>
  <input
    className="form-control"
    id="searchInput"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn btn-primary ms-2"
  >
    Search
  </button>
</form>
`;

exports[`Tests in <SearchBar /> Should be displayed correctly 1`] = `
<form
  className="d-flex"
  onSubmit={[Function]}
>
  <input
    className="form-control"
    id="searchInput"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="btn btn-primary ms-2"
  >
    Search
  </button>
</form>
`;
